### DISCUSSION TITLE
Include current progress in the mdapi convert option

### DISCUSSION THREAD

**Discussion by mnunezdm** [forcedotcom/cli#2261] [Ideas]
First of all, congratulations in the performance upgrade, with the previous version (I was using sonar@1.0.0) doing the conversion in a big project (almost 20k metadata components) it took 10+ minutes. Using the beta version (1.5.3) it only takes 60 seconds

**Is your feature request related to a problem? Please describe.**

When performing a conversion, it only shows the total number of metadata components to be converted, not the actual progress.

**What are you trying to do**

Give the user the actual conversion progress.

**Describe the solution you'd like**

I would love to have a progress bar to show the actual conversion progress

**Describe alternatives you've considered**

na

**Additional context**

na

[a comment by an automated account is not shown]

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-10219182](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000Rs90MYAR/view)

**Comment by shetzel** [comment]
That's great to hear about the perf gain!  I think a progress bar would be a great addition as well, nice idea.

**Comment by mshanemc** [comment]
Deploys are easy because the mdapi responses contain the stats, so they update as we poll).

The stream-based architecture of SDR makes this more challenging.  We'd have to add an eventEmitter to the `MetadataConverter` [here](https://github.com/forcedotcom/source-deploy-retrieve/blob/c8a1191550a93815460722fbb9e92b4fdb5717be/src/convert/metadataConverter.ts#L65-L66) to listen to events from the writeStream and pass them out to the command so it can do the progress bar.
